driver-core/src/main/com/mongodb/internal/operation/CreateCollectionOperation.java

```diff
@@ -361,6 +366,27 @@ private BsonDocument getCreateCollectionCommand() {
         return document;
     }
 
+    private void checkEncryptedFieldsSupported(final ConnectionDescription connectionDescription) throws MongoException {
+        if (encryptedFields != null && serverIsLessThanVersionSevenDotZero(connectionDescription)) {
+            throw new MongoClientException("Driver support of Queryable Encryption is incompatible with server."
+                    + " Upgrade server to use Queryable Encryption.");
+        }
+    }
+
+    /**
+     * @return {@code true} iff the {@linkplain #checkEncryptedFieldsSupported(ConnectionDescription) check} was successful.
+     * The {@code callback} is completed (with a failed result) iff the check was not successful.
+     */
+    private boolean checkEncryptedFieldsSupported(final ConnectionDescription connectionDescription, final SingleResultCallback<Void> callback) {
+        try {
+            checkEncryptedFieldsSupported(connectionDescription);
+            return true;
+        } catch (Exception e) {
```

Comment:
[Optional] 

I noticed that checkEncryptedFieldsSupported method is declared to throw MongoException. However, the method catches Exception, which covers all types of exceptions including both checked and runtime exceptions. This could potentially catch more than necessary. Considering narrowing it down to RuntimeException since if a checked exception is thrown down the road later, the compiler will bring it to attention to treat it as special case. Otherwise, it might be caught unexpectedly.

[RSPEC-2221](https://rules.sonarsource.com/java/RSPEC-2221)

Reply:
Since it is designed for a callback to handle all exceptions and log errors for the user to see, therefore, disregard comment above.

Reply:
> Considering narrowing it down to RuntimeException since if a checked exception is thrown down the road later, the compiler will bring it to attention to treat it as special case.

I would have loved if we could do this, and actually, this is what I was doing previously. However, then https://jira.mongodb.org/browse/JAVA-4575 happened (which is really, Kotlin's fault). To avoid thinking too much each time whether a method may sneakily throw a checked exception due to executing external code, we decided to catch `Exception` even though it's weird if one thinks only about Java.